daemon-scheduler/pkg/deployment/environment.go

```diff
@@ -50,8 +50,15 @@ type Environment interface {
 	// UpdateDeployment replaces an existing deployment in the environment with the
 	// provided one if a deployment with the provided ID already exists
 	UpdateDeployment(ctx context.Context, environment types.Environment, deployment types.Deployment) (*types.Environment, error)
-	//GetCurrentDeployment returns the deployment which needs to be used for starting tasks
+	//GetCurrentDeployment returns the deployment which needs to be used for starting tasks, i.e.
+	// in-progress deployment if one exists, otherwise the latest completed deployment.
 	GetCurrentDeployment(ctx context.Context, name string) (*types.Deployment, error)
+	// GetInProgressDeployment returns the in-progress deployment for the environment.
+	// There should be no more than one in progress deployments in an environment.
+	GetInProgressDeployment(ctx context.Context, name string) (*types.Deployment, error)
+	// GetDeploymentsSortedReverseChronologically returns a list of deployments reverse-ordered by start time,
+	// i.e. lastest deployment first
+	GetDeploymentsSortedReverseChronologically(ctx context.Context, name string) ([]types.Deployment, error)
```

Comment:
Should we use ListDeployments... instead? To keep it consistent with ListEnvironments.